src/main/kotlin/no/nav/tiltakspenger/arena/repository/ArenaSakDTO.kt

```diff
@@ -18,7 +18,32 @@ data class ArenaSakDTO(
         get() = ihtVedtak.first { it.vedtakType == ArenaVedtakType.O }.mottattDato
     val fagsystemSakId: String
         get() = aar.toString() + lopenrSak
+
+    fun harVedtakMedÅpenPeriode(): Boolean =
+        this.ihtVedtak.any { it.isVedtaksperiodeÅpen() }
+
+    fun førsteFomVedtaksperiodeIsBeforeOrEqualTo(date: LocalDate): Boolean =
+        this.ihtVedtak
+            .mapNotNull { it.fomVedtaksperiode }
+            .minOrNull()
+            ?.let { it == date || it.isBefore(date) } ?: true
+
+    fun sisteVedtakMedLukketPeriodeIsAfterOrEqualTo(date: LocalDate): Boolean =
+        this.ihtVedtak
+            .mapNotNull { it.tomVedtaksperiode }
+            .maxOrNull()
+            ?.let { it == date || it.isAfter(date) } ?: true
+
+    fun harVedtak(): Boolean = this.ihtVedtak.isNotEmpty()
 }
+
+fun List<ArenaSakDTO>.kunSakerMedVedtakInnenforPeriode(fom: LocalDate, tom: LocalDate) =
```

Comment:
Er ikke så kjent med strukturen på extension functions, men seturnerer denne en Boolean? Kan man i så fall  putte en ': Boolean’ for å gjøre det mer tydelig?

Reply:
Den, den returnerer en liste. Men jeg er helt enig, jeg liker eksplisitte returtyper. Jeg har lagt det til her og også en del andre steder nå.